lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -4,9 +4,31 @@ module ActiveModel
   class Serializer
     class Adapter
       class JsonApi < Adapter
+        @root = :data
+        DEFAULT_ATTRIBUTES = [:type, :id]
+
+        def self.params(permitted, associations)
+          relationships = {}
+          associations.each do |assoc|
+             relationships[assoc] = {}
+             relationships[assoc][@root] = DEFAULT_ATTRIBUTES
+           end
+
+          return :type, {attributes: permitted}, {relationships: relationships}
```

Comment:
`:id` should be allowed here, in order to handle client-generated IDs (http://jsonapi.org/format/#crud-creating).